src/libraries/System.Runtime.Loader/ref/System.Runtime.Loader.cs

```diff
@@ -12,6 +12,12 @@ public static partial class AssemblyExtensions
         public unsafe static bool TryGetRawMetadata(this System.Reflection.Assembly assembly, out byte* blob, out int length) { throw null; }
         public static void ApplyUpdate(Assembly assembly, ReadOnlySpan<byte> metadataDelta, ReadOnlySpan<byte> ilDelta, ReadOnlySpan<byte> pdbDelta) { throw null; }
     }
+    [System.AttributeUsage(System.AttributeTargets.Assembly, AllowMultiple = true)]
+    public sealed class MetadataUpdateHandlerAttribute : System.Attribute
+    {
+        public MetadataUpdateHandlerAttribute([System.Diagnostics.CodeAnalysis.DynamicallyAccessedMembersAttribute(System.Diagnostics.CodeAnalysis.DynamicallyAccessedMemberTypes.All)] System.Type handlerType) { }
+        public System.Type HandlerType { get { throw null; } }
```

Comment:
Typically if the ctor annotates the param as `DynamicallyAccessedMembers`, you will also want to annotate the property.

I assume there is going to be some caller that loops through all the `MetadataUpdateHandlerAttribute` instances, and starts calling methods on the `HandlerType` (using Reflection?). If that is the case, that caller is going to get ILLink warnings that methods on the `HandlerType` might be trimmed. So adding the attribute to the property will pre-emptively fix that.

Reply:
Ok.  I don't know if the caller is going to use the property or the ctor argument, and I don't expect the caller (the hot reload agent) to have trimming enabled, but it doesn't hurt to put the attribute in both places.